### DISCUSSION TITLE
为什么fun的参数通常是reference，而script的参数通常是value

### DISCUSSION THREAD

**Discussion by livelybug** [NonceGeek/Move-Camp#158] [Q&A] [answered]
如题，这只是约定俗成吗？
```rust
     public fun init(account: signer){
        move_to(&account, Counter{value:0});
     }
     public(script) fun init_counter(account: signer){
        Self::init(account)
     }

```

**Comment by hi-chenerge** [comment]
这个问题应该是写反了，应该是这样

     public fun init(account: &signer){
        move_to(account, Counter{value:0});
     }

     public(script) fun init_counter(account: signer){
        Self::init(&account)
     }

**Reply by livelybug** [reply]
> 这个问题应该是写反了，应该是这样
> 
> ```
>  public fun init(account: &signer){
>     move_to(account, Counter{value:0});
>  }
> 
>  public(script) fun init_counter(account: signer){
>     Self::init(&account)
>  }
> ```

呃 好像是写反了，
问题的意思应该有说清楚：fun用refernce, script用value是不是约定俗成

**Comment by mason0510** [comment, marked as answer]
没有。 看需要

**Reply by hi-chenerge** [reply]
能不能具体说一下呢
